### PR TITLE
Add LayerTree

### PR DESCRIPTION
**Pull request by oterral**
@edellucien

### CODE REVIEW

**Comment by mhaertwig** [comment]
Looks nice so far! What about centralizing the layer interaction behavior in a `LayerService` (see spec). Is it still planned?

**Comment by oterral** [comment]
of course we need to discuss that. @edellucien begins the work on this.For now, the layer Tree is totally agnostic from  the map. And we should keep that way if we can, it will better for maintenance.

**Comment by oterral** [comment]
Ready to review. I will add the stuff to have a configurable css theme in another PR.

File: src/components/layertree/LayerTree.js [hunk 1]
```diff
@@ -0,0 +1,389 @@
...
+          isExpanded: PropTypes.bool,
+        }),
+      }),
+    ).isRequired,
+  }),
+
+  /**
+   * CSS class to apply on each item.
+   */
+  classNameItem: PropTypes.string,
```
Comment (mhaertwig) [inline src/components/layertree/LayerTree.js]: For consistency (see e.g. Button.js) please use `className`

**Comment by mhaertwig** [comment]
Nice job so far. Thanks! Some comments:
- The behavior in barrier free mode should be the same as on https://maps.trafimage.ch
- Please make the example a bit nicer:
  - Show layer tree on map
  - Use some better example layers with their real names
  - The interactive stationplan layer isn't really a good example layer. Please use something else (e.g. OSM).
  - Remove the Japanese flag from the examples.

File: src/components/layertree/LayerTree.js (hunk 1 above)
Comment (oterral) [reply src/components/layertree/LayerTree.js]: `className`, it  feels  it's for the parent component.  but it's for the item.

**Comment by oterral** [comment]
@mhaertwig  for better examples, @edellucien is on it with the LayerService.

**Comment by oterral** [comment]
I close this in favor of https://github.com/geops/react-spatial/pull/30

**Pull request closed without merging**